linux/mcompile/keymap/mcompile.cpp

```diff
@@ -412,13 +415,14 @@ void KMX_ConvertDeadkey(LPKMX_KEYBOARD kbd, KMX_WORD vk_US, KMX_DWORD shift, KMX
   KMX_FDeadkeys.push_back(KMX_deadkeyMapping);  // dkid, vk, shift);   // I4353
   KMX_AddDeadkeyRule(kbd, dkid, vk_US, shift);
 
-  KMX_GetDeadkeys(dk_Table, deadkey, pdk = deadkeys, keymap);  // returns array of [usvk, ch_out] pairs
+  KMX_GetDeadkeys(dk_Table, deadkey, vec_deadkeys, keymap);  // returns vector of [usvk, ch_out] pairs
 
-  while (*pdk) {
+  int n=0;
+  while (n < (int)vec_deadkeys.size()) {
```

Comment:
That is true, and it works. However, using `vec_deadkeys.size() - 2` doesn't change anything in the way the current code works, but it would be a sign of defensive programming.